src/lib/hooks/useFilteredFields.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect } from 'react';
+import { EProjectFields } from '../consts/projects';
+
+export const useFilteredFields = (
+  watchedFields: any,
+  setActiveFilters: any,
+) => {
+  useEffect(() => {
+    const activeFilterFields: Partial<Record<EProjectFields, boolean>> = {};
+    Object.entries(watchedFields).forEach(([field, { value }]: any) => {
```

Comment:
no any

Reply:
here is an example version of how to solve the types:
```ts
import { useEffect } from 'react';

type IFieldValue = string[] | Record<string, string>;

type IField = { value: IFieldValue };

type IWatchFields<F extends string = string> = {
  [key in F]: IField;
};

type IActiveFiltes<F extends string = string> = Record<F, boolean>;

const getEntries = <T extends {}>(obj: T) =>
  Object.entries(obj) as Array<[keyof T, T[keyof T]]>;

export const useFilteredFields = <F extends string = string>(
  watchedFields: IWatchFields<F>,
  setActiveFilters: (filters: IActiveFiltes<F>) => void,
) => {
  useEffect(() => {
    const activeFilterFields: IActiveFiltes<F> = {} as IActiveFiltes<F>;
    getEntries(watchedFields).forEach(([field, { value }]) => {
      if (
        (Array.isArray(value) && value.length > 0) ||
        (typeof value === 'object' && Object.values(value).some(Boolean))
      ) {
        activeFilterFields[field] = true;
      }
    });
    setActiveFilters({ ...activeFilterFields });
  }, [watchedFields, setActiveFilters]);
};
```

in a function scope you must provide the correct types , tough we can use type assertion in case typescript cannot follow the logic and we can convince him what are the actual types...